probe_scraper/parsers/utils.py

```diff
@@ -24,3 +24,15 @@ def get_major_version(version):
     :return: a string containing the leftmost number before the first dot
     """
     return version.split(".")[0]
+
+
+def add_source_url(v, repo_url, commit_hash):
```

Comment:
```suggestion
def get_source_url(defined_in, repo_url, commit_hash):
```

I think it's probably clearer if we make this function just return a source URL and pass in the `defined_in` part of Glean Parser's output. You can then do the various munging inline (it's a little bit of duplication, but not too much and I think worth it for the additional clarity).

Reply:
I might call this data structure `glean_definition`

Reply:
What Will is suggesting is to avoid side effects, which is a good pattern that really evolves from functional programming.